scripts/transports.lua

```diff
@@ -3,6 +3,10 @@ if GG.TransportAllowed then
 end
 
 function GG.TransportAllowed(passengerID)
+	local disarmed = Spring.GetUnitRulesParam(unitID, "disarmed")
+	if (disarmed and disarmed == 1) then
+		return false
+	end
```

Comment:
operator == works correctly with `nil`, could just do
```suggestion
	if (spGetUnitRulesParam(unitID, "disarmed") == 1) then
		return false
	end
```